backend/src/exercise/websocket-handler/secure-on.spec.ts

```diff
@@ -0,0 +1,22 @@
+import { createTestEnvironment, sleep } from '../../../test/utils';
+
+describe('secure on', () => {
+    const environment = createTestEnvironment();
+    it('does not crash in production environment when fed with invalid data', async () => {
+        const originalEnv = process.env.NODE_ENV;
+        // Errors are only catched in production mode
+        process.env.NODE_ENV = 'production';
+        try {
+            await environment.withWebsocket(async (socket) => {
+                socket.insecureEmit('getState', 'any');
+                // wait a bit so that the error can occur
+                await sleep(1000);
+                // Check whether the server is still available
+                await environment.httpRequest('get', '/api/health').expect(200);
+            });
+        } finally {
+            process.env.NODE_ENV = originalEnv;
+        }
```

Comment:
```suggestion
        await environment
            .withWebsocket(async (socket) => {
                socket.insecureEmit('getState', 'any');
                // wait a bit so that the error can occur
                await sleep(1000);
                // Check whether the server is still available
                await environment.httpRequest('get', '/api/health').expect(200);
            })
            .finally(() => {
                process.env.NODE_ENV = originalEnv;
            });
```

I would have thought this is shorter but it's actually the same number of lines.

You can decide what's better.

Either way, approved.

Reply:
I think I like your version better. Visually it better ties the finally to the Promise instead of the await imo.